corehq/apps/users/views/__init__.py

```diff
@@ -963,6 +963,12 @@ def _delete_user_role(domain, role_data):
     except UserRole.DoesNotExist:
         raise Http404
 
+    if role.is_commcare_user_default:
+        raise ValueError(_(
+            "Unable to delete role '{role}'. "
+            "This role is the default role for Mobile Users and can not be deleted.",
+        ).format(role=role_data["name"]))
```

Comment:
random thought: if we do eventually get to making roles required for all users, we could allow this to be deleted.

Reply:
I had also thought that we could add a role select box when creating mobile users which would then also mean we could drop the concept of a 'mobile worker default role' (once all users have a role set).